tools/serverpod_cli/test/generator/dart/server_code_generator/custom_class_test.dart

```diff
@@ -68,5 +69,53 @@ void main() {
             'The fromJson method should pass data as dynamic to CustomClass.fromJson but doesn\'t.',
       );
     });
+
+    test(
+      'then toJsonForProtocol method should correctly serialize customClassField by calling the appropriate toJson method',
+      () {
+        var toJsonForProtocolMethod =
+            CompilationUnitHelpers.tryFindMethodDeclaration(
+          maybeClassNamedExample!,
+          name: 'toJsonForProtocol',
+        );
+
+        var toJsonForProtocolCode = toJsonForProtocolMethod!.toSource();
+
+        var expectedCode =
+            '@override Map<String, dynamic> toJsonForProtocol() {'
+            'return {'
+            "'customClassField' : customClassField is _i1.ProtocolSerialization "
+            '? (customClassField as _i1.ProtocolSerialization).toJsonForProtocol() : customClassField.toJson()'
+            '};'
+            '}';
```

Comment:
Suggestion: Break out this to multiple expects for different parts of the code.

The expect here will be to sensitive to change. If the "style" changes this test might fail for irrelevant reasons. Please validate the important parts separately with different expects.

Reply:
Could also be different "tests" to slim down what each expect is testing.